File: php/2-fibonacci.php
<?php
$time_start = microtime(true);

$max = 4000000;
$fibonacciSequence = [1];
$fibonacci = function (int $previous = 1) use (&$fibonacci, &$fibonacciSequence, $max): array {
    end($fibonacciSequence);
    $current = current($fibonacciSequence);
    if ($max > $current) {
        $fibonacciSequence[] = $previous + $current;
        $fibonacci($current);
    }
    return $fibonacciSequence;
};

$evenFibonacci = array_filter(
    $fibonacci(),
    function ($fibonacci) {
        return (0 === $fibonacci % 2);
    }
);

$sumOfEvenFibonacci = array_reduce(
    $evenFibonacci,
    function ($sum, $item) {
        $sum += $item;
        return $sum;
    }
);

echo $sumOfEvenFibonacci;

$time_end = microtime(true);

//dividing with 60 will give the execution time in minutes otherwise seconds
$execution_time = ($time_end - $time_start)/60;

//execution time of the script
echo '<b>Total Execution Time:</b> '.number_format((float) $execution_time, 10).' Mins';

File: python/2-fibonacci.py
fibonacci = [1, 2]
max_number = 4000000000000
even_fibonacci = []
pointer = 1
fibonacci_number = 0

while fibonacci_number < max_number:
    fibonacci_number = fibonacci[pointer] + fibonacci[pointer - 1]
    fibonacci.append(fibonacci_number)
    pointer = pointer + 1

sum_of_even_fibonacci = sum(list(filter(lambda fibonacci_even_number: 0 == fibonacci_even_number % 2, fibonacci)))
print(sum_of_even_fibonacci)
